wandb/sdk/wandb_run.py

```diff
@@ -592,21 +592,23 @@ def __init__(
     ) -> None:
         # pid is set, so we know if this run object was initialized by this process
         self._init_pid = os.getpid()
+        self._settings = settings
+
+        if settings._noop:
+            return
```

Comment:
skipping init here means many attributes are not set, which can cause problems for code which uses variables, e.g. `.entity`. Is there a reason `_init` can't be called to set default values?